src/Workspaces/Core/Portable/SourceGeneration/IRemoteSourceGenerationService.cs

```diff
@@ -12,6 +12,12 @@
 
 namespace Microsoft.CodeAnalysis.SourceGeneration;
 
+[DataContract]
+internal readonly record struct RegularCompilationTrackerSourceGenerationInfo(
+    [property: DataMember(Order = 0)] SourceGeneratedDocumentIdentity DocumentIdentity,
+    [property: DataMember(Order = 1)] SourceGeneratedDocumentContentIdentity ContentIdentity,
+    [property: DataMember(Order = 2)] DateTime GenerationDateTime);
```

Comment:
extracted out a helper type as this was just causing a massively long signature line.